virtual/pandoc/pandoc-0.ebuild

```
@@ -0,0 +1,15 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="Virtual for pandoc"
+
+LICENSE=""
+SLOT="0"
+KEYWORDS="amd64 ~arm64 ~riscv ~x86"
+
+BDEPEND=""
+RDEPEND="|| ( app-text/pandoc-cli app-text/pandoc-bin[pandoc-symlink] )"
```

```suggestion
RDEPEND="|| ( app-text/pandoc-cli app-text/pandoc-bin[pandoc-symlink] <app-text/pandoc-3 )"
```

I'm thinking this would be a small improvement that would allow for installing pandoc from `::gentoo` if someone really wanted to.

1. `app-text/pandoc` is now named `dev-haskell/pandoc` in `::haskell`, so there shouldn't be any danger of accidentally pulling in the library instead of the executable from `::haskell`.
2. The split from `pandoc` to `pandoc`/`pandoc-cli` happened after `pandoc-3` was released, so anything before version 3 will be a pre-split version.

---

This should also help with syncing these ebuilds to `::gentoo` in the future.

It might be a good idea to soft-block `app-text/pandoc` ⟺ `app-text/pandoc-cli`/`dev-haskell/pandoc` as well, that way users will be able to upgrade from the stabilized `app-text/pandoc-2.18-r2::gentoo` to the new split packages.

---

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;I wasn't quite sure how to handle `app-text/pandoc::gentoo` vs. `app-text/pandoc-cli::haskell`.  This change makes sense, so I'll also incorporate it, too.   (As for a soft-block, though, I haven't implemented one in an ebuild yet.  I'll look for what documentation exists on doing that.)

---

It's documented here:

https://devmanual.gentoo.org/general-concepts/dependencies/index.html#blockers

So, e.g. you would add this to `app-text/pandoc-cli` and `dev-haskell/pandoc`:

```ebuild
RDEPEND="!app-text/pandoc"
```

---

We should also add complementary blockers to the `app-text/pandoc` ebuild in `::gentoo`, which will require [opening a PR](https://github.com/gentoo/gentoo/pulls) there.

---

That will also require syncing `app-text/pandoc-cli` and `dev-haskell/pandoc` to `::gentoo`, which would pull in a bunch of other dependencies. It would make sense to do a big sync PR there. I'll go ahead and start working on it.

---

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Thanks for the doucmentation pointer.  I'll let you handle the big PR for syncing with `::gentoo` upstream, if I may.

---

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will soft-blocking `app-text/pandoc::gentoo` on `app-text/pandoc-cli` and `dev-haskell/pandoc` be included in your sync PR, or would it make sense to do that separately if it can land earlier?

---

No problem. Syncing with `::gentoo` is pretty complex.

---

> Will soft-blocking `app-text/pandoc::gentoo` on `app-text/pandoc-cli` and `dev-haskell/pandoc` be included in your sync PR, or would it make sense to do that separately if it can land earlier?

My understanding is that blocks need to be bi-directional, so your changes here will be included in the PR to `::gentoo`. Both PRs will need to be merged at roughly the same time.

---

>> > &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Will soft-blocking `app-text/pandoc::gentoo` on `app-text/pandoc-cli` and `dev-haskell/pandoc` be included in your sync PR, or would it make sense to do that separately if it can land earlier?
> 
> My understanding is that blocks need to be bi-directional, so your changes here will be included in the PR to `::gentoo`. Both PRs will need to be merged at roughly the same time.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;I was referring to the corresponding change in `app-text/pandoc::gentoo` to add '`!app-text/pandoc-cli`' and '`!dev-haskell/pandoc`' to its '`RDEPENDS`' variable.  This PR of mine here only adds '`RDEPENDS="!app-text/pandoc"`' to '`app-text/pandoc-cli`' and `'dev-haskell`' here in '`::haskell`.'

---

Right, don't worry about changing `app-text/pandoc::gentoo`. I'll get the soft-block added to it in my PR.

---

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Awesome; thank you very much!